tests/conftest.py

```diff
@@ -84,16 +71,50 @@ def playwright():
         yield p
 
 
-@pytest.fixture(params=["chromium", "firefox", "webkit"])
-def browser(playwright: Playwright, request: SubRequest, server: Generator[Any, Any, Any]) -> Generator[Page, Any, Any]:
+@pytest.fixture(params=["chromium"])  # lets start with 1 browser for now, we can add more later
+def browser(
+    playwright: Playwright, request: SubRequest, run_server: Generator[str, Any, Any]
+) -> Generator[Page, Any, Any]:
     browser = getattr(playwright, request.param).launch(headless=True)
-    context = browser.new_context(base_url=server)
+    context = browser.new_context(base_url=run_server)
     page = context.new_page()
-    wait_for_server_ready(server)
+
+    transport = httpx.HTTPTransport(retries=5)
+    with httpx.Client(transport=transport, verify=False) as client:  # noqa: S501
+        client.get(f"{run_server}/", timeout=0.3)
+
     yield page
     browser.close()
 
 
 @pytest.fixture()
-def db(get_session: Generator[Session, Any, Any]):
-    return DatabaseTestUtils(get_session)
+def db(get_session: Session) -> Generator[DatabaseTestUtils, None, None]:
+    database = DatabaseTestUtils(get_session)
+    yield database
+    del database
```

Comment:
The del command seems to mark it for deletion, but it may not happen in the same run/session. I get strange 'object is not in session' errors on the models. It does work if you call it explicitly, so I'd change it to a callable function instead of cleanup.

```suggestion
    database.reset()
```

Reply:
I am not a big fan of making it a custom function. It would be great if the objects are autoamticly removed on destruction of the object.

Reply:
i will resolve it in the destructor

Reply:
@berrydenhartog the error seems to occur when pytest_sessionfinish runs before the 'del' is executed. The database tables have been cleared already then. As the goal is the delete the objects / tables, that is what does happen, so okayish with it.

Reply:
Interresting find. i did not expect the pytest_sessionfinish to influence that